packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -96,45 +97,67 @@ function useInsertionPoint( {
 		hideInsertionPoint,
 	} = useDispatch( 'core/block-editor' );
 
-	const onInsertBlocks = ( blocks, meta ) => {
-		if (
-			! isAppender &&
-			selectedBlock &&
-			isUnmodifiedDefaultBlock( selectedBlock )
-		) {
-			replaceBlocks( selectedBlock.clientId, blocks, null, null, meta );
-		} else {
-			insertBlocks(
-				blocks,
-				destinationIndex,
-				destinationRootClientId,
-				selectBlockOnInsert,
-				meta
-			);
-		}
+	const onInsertBlocks = useCallback(
+		( blocks, meta ) => {
+			if (
+				! isAppender &&
+				selectedBlock &&
+				isUnmodifiedDefaultBlock( selectedBlock )
+			) {
+				replaceBlocks(
+					selectedBlock.clientId,
+					blocks,
+					null,
+					null,
+					meta
+				);
+			} else {
+				insertBlocks(
+					blocks,
+					destinationIndex,
+					destinationRootClientId,
+					selectBlockOnInsert,
+					meta
+				);
+			}
 
-		if ( ! selectBlockOnInsert ) {
-			// translators: %d: the name of the block that has been added
-			const message = _n(
-				'%d block added.',
-				'%d blocks added.',
-				blocks.length
-			);
-			speak( message );
-		}
+			if ( ! selectBlockOnInsert ) {
+				// translators: %d: the name of the block that has been added
+				const message = _n(
+					'%d block added.',
+					'%d blocks added.',
+					blocks.length
+				);
+				speak( message );
+			}
 
-		if ( onSelect ) {
-			onSelect();
-		}
-	};
+			if ( onSelect ) {
+				onSelect();
+			}
+		},
+		[
+			isAppender,
+			selectedBlock,
+			isUnmodifiedDefaultBlock,
+			replaceBlocks,
+			insertBlocks,
+			destinationRootClientId,
+			destinationIndex,
+			selectBlockOnInsert,
+			onSelect,
+		]
```

Comment:
Should we remove the static imports from this list (isUnmodifiedDefaultBlock seems to be static)

Reply:
Nice catch, thanks!